### PR TITLE
Allow escaping of ? in fragment string

### PR DESCRIPTION
**Pull request by ctlevi**
Fragment strings can escape question mark with a backslash

### CODE REVIEW

File: lib/ecto/query/builder.ex [hunk 1]
```diff
@@ -203,6 +203,12 @@ defmodule Ecto.Query.Builder do
     error! "`#{Macro.to_string(other)}` is not a valid query expression"
   end
 
+  defp split_binary(query), do: split_binary(query, "")
+  defp split_binary(<<>>, consumed), do: [consumed]
+  defp split_binary(<<63, rest :: binary >>, consumed), do: [consumed | split_binary(rest, "")]
```
Comment (josevalim) [inline lib/ecto/query/builder.ex]: You can use `??` (63) and `?\\` (92) :)

**Comment by josevalim** [comment]
That's awesome, I have added one tiny comment and we are ready to go!

File: lib/ecto/query/builder.ex (hunk 1 above)
Comment (ctlevi) [inline lib/ecto/query/builder.ex]: Very cool! I must have missed that in the elixir tutorial.

**Comment by josevalim** [comment]
:heart:

**Pull request merged**